docs/src/creating-packages.md

```diff
@@ -366,6 +371,14 @@ If one considers `PlottingContourExt` as a completely separate package, it could
 However, for extensions, it is ok to assume that the extension owns the methods in its parent package.
 In fact, this form of type piracy is one of the most standard use cases for extensions.
 
+Conversely, although it is possible, defining new functions or types in the extension is not recommended.
+If your use case requires it, you can call `Base.get_extension(Plotting, :PlottingContourExt)` to retrieve the extension module.
+From there, any name such as `MyFancyContourPlot` can be retrieved with the standard module prefixing syntax.
+If you use it often in a third party package, you may want to shorten it as follows:
+```julia
+const MyFancyContourPlot = Base.get_extension(Plotting, :PlottingContourExt).MyFancyContourPlot
+```
+
```

Comment:
Should we also mention here the "recipe" where the parent package provides a way to access things within its extension?

Like
```julia
# within the Plotting module:
function get_ext()
    ext = Base.get_extension(@__MODULE__, :PlottingContourExt)
    if ext === nothing
        error("extension not loaded...")
    else
        return ext  # or ext.something_defined_in_ext
    end
end
```

Reply:
I wasn't aware of this! Is it widely used?